resolwe_bio/processes/import_data/sra_file.py

```diff
@@ -0,0 +1,403 @@
+                    'TOPHRED33',
+                ]()
+                os.rename('reformated.fastq.gz', f'{reads_name}.fastq.gz')
+
+            elif encoding != "Sanger / Illumina 1.9\n":
+                self.error(
+                    "Only Sanger / Illumina 1.9 / llumina 1.5 / Illumina 1.3 encoding is "
+                    "supported."
+                )
+
+        fastqc = glob.glob('*_fastqc.zip')
+
+        outputs.fastqc_url = fastqc_url
+        outputs.fastqc_archive = fastqc
+
+
+class ImportSraPaired(Process):
+    """Import paired-end reads from SRA.
+
+    Import paired-end reads from Sequence Read Archive (SRA) via an SRA accession number.
+    SRA stores raw sequencing data and alignment information from high-throughput
+    sequencing platforms.
+
+    """
+    slug = 'import-sra-paired'
+    name = 'SRA data (paired-end)'
+    process_type = 'data:reads:fastq:paired'
+    version = '1.0.0'
+    category = 'Import'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.RAW
+    entity = {
+        'type': 'sample',
+        'descriptor_schema': 'sample',
+    }
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/common:1.3.1'
+            }
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 1024,
+            'network': True,
+        },
+    }
+    data_name = '{{ sra_accession|first }}'
+
+    class Input:
+        """Input fields to process ImportSraPaired."""
+        sra_accession = ListField(StringField(), label='SRA accession(s)')
+        show_advanced = BooleanField(label='Show advanced options', default=False)
+
+        class Advanced:
+            """Advanced options."""
+            min_spot_id = IntegerField(label='Minimum spot ID', required=False)
+            max_spot_id = IntegerField(label='Maximum spot ID', required=False)
+            min_read_len = IntegerField(label='Minimum read length', required=False)
+            clip = BooleanField(label='Clip adapter sequences', default=False)
+            aligned = BooleanField(label='Dump only aligned sequences', default=False)
+            unaligned = BooleanField(label='Dump only unaligned sequences', default=False)
+
+        advanced = GroupField(Advanced, label='Advanced options', hidden='!show_advanced')
+
+    class Output:
+        """Output fields to process ImportSraSingle."""
+        fastq = ListField(FileField(), label='Reads file (mate 1)')
+        fastq2 = ListField(FileField(), label='Reads file (mate 2)')
+        fastqc_url = ListField(
+            FileHtmlField(),
+            label='Quality control with FastQC (Upstream)',
+        )
+        fastqc_url2 = ListField(
+            FileHtmlField(),
+            label='Quality control with FastQC (Downstream)',
+        )
+        fastqc_archive = ListField(FileField(), label='Download FastQC archive (Upstream)')
```

Comment:
Do we want to use consistent mate1/mate2 annotation here in place of upstream/downstream?

Reply:
mate1/mate2 is the preferred way, yes!